### PR TITLE
ignore LdcEqualsLdd and let all kernel support ldc != ldd.

### PR DESCRIPTION
**Pull request by TonyYHsieh**
ignore LdcEqualsLdd and let all kernel support ldc != ldd.

This change let Tensile inherently support ldc != ldd.
So caller, ex. rocblas, doesn't have to handle ldc!=ldd.

We have compare performance before and after change.
image
[ldd.xlsx](https://github.com/ROCmSoftwarePlatform/Tensile/files/5462818/ldd.xlsx)

### CODE REVIEW

File: Tensile/Configs/rocblas_dgemm_bufferload_limit.yaml [hunk 1]
```diff
@@ -1,15 +1,18 @@
+ # the problems sizes in this test is extremely
+ # skip some old cards, not sure if it's too large to be allocated
+TestParameters:
+  marks: [skip-gfx900, skip-gfx906, skip-gfx1010, skip-gfx1011]
+
```
Comment (solaslin) [inline Tensile/Configs/rocblas_dgemm_bufferload_limit.yaml:5]: Remove this TestParameters block which is only for Tensile tox test

File: Tensile/Configs/rocblas_sgemm_bufferload_limit.yaml [hunk 2]
```diff
@@ -1,10 +1,13 @@
+ # the problems sizes in this test is extremely
+ # skip some old cards, not sure if it's too large to be allocated
+TestParameters:
+  marks: [skip-gfx900, skip-gfx906, skip-gfx1010, skip-gfx1011]
+
```
Comment (solaslin) [inline Tensile/Configs/rocblas_sgemm_bufferload_limit.yaml:5]: Remove this TestParameters block which is only for Tensile tox test

File: Tensile/KernelWriterAssembly.py [hunk 3]
```diff
@@ -1127,24 +1128,12 @@ def initKernel(self, kernel, tPA, tPB ):
...
-    #     Dexternal = Cexternal (= DestDataType) and is the final gemm result
-    #
-    # - For GSU: the "current" kernel is a GSU-kernel,
-    #     this kernel returns a temp buffer with same type as Cinternal.
-    #     Later, another kernel will accumulate this buffer
-    #     and convert the final result to Cexternal (= DestDataType) as the gemm result
-    self.bpeDexternal = int(self.bpr * kernel["ProblemType"]["DestDataType"].numRegisters())
-
-    # TODO- Currently, Cexternal, Cinteral, Dexternal are kind of confusing, need to clean up
-
```
Comment (solaslin) [inline Tensile/KernelWriterAssembly.py:1141]: Dexternal is removed, but could we still put the original comments here to account for "Cexternal" ?
```
    # Cexternal = the "current" kernel output type,
    # - default: the "current" kernel is a non-GSU-kernel,
    #     Cexternal (= DestDataType) and is the final gemm result
    #
    # - For GSU: the "current" kernel is a GSU-kernel,
    #     this kernel returns a temp buffer with same type as Cinternal.
    #     Later, another kernel will accumulate this buffer
    #     and convert the final result to DestDataType as the gemm result
```

**Review by solaslin — CHANGES_REQUESTED** [review]

File: Tensile/KernelWriterAssembly.py (hunk 3 above)
Comment (TonyYHsieh) [reply Tensile/KernelWriterAssembly.py:1141]: done

File: Tensile/Configs/rocblas_sgemm_bufferload_limit.yaml (hunk 2 above)
Comment (TonyYHsieh) [reply Tensile/Configs/rocblas_sgemm_bufferload_limit.yaml]: done

File: Tensile/Configs/rocblas_dgemm_bufferload_limit.yaml (hunk 1 above)
Comment (TonyYHsieh) [reply Tensile/Configs/rocblas_dgemm_bufferload_limit.yaml]: done

**Review by solaslin — APPROVED** [review]

File: Tensile/Tests/extended/bufferload_offset/rocblas_dgemm_bufferload_limit.yaml [hunk 4]
```diff
@@ -1,3 +1,8 @@
+ # the problems sizes in this test is extremely
```
Comment (rosenrodt) [inline Tensile/Tests/extended/bufferload_offset/rocblas_dgemm_bufferload_limit.yaml:1]: Skipped tests on those platforms by mistake?

File: Tensile/Tests/extended/bufferload_offset/rocblas_sgemm_bufferload_limit.yaml [hunk 5]
```diff
@@ -1,3 +1,8 @@
+ # the problems sizes in this test is extremely
+ # skip some old cards, not sure if it's too large to be allocated
+TestParameters:
```
Comment (rosenrodt) [inline Tensile/Tests/extended/bufferload_offset/rocblas_sgemm_bufferload_limit.yaml:3]: Same here. If CI doesn't complain in the past then there's no need to skip test

File: Tensile/Tests/extended/bufferload_offset/rocblas_dgemm_bufferload_limit.yaml (hunk 4 above)
Comment (TonyYHsieh) [reply Tensile/Tests/extended/bufferload_offset/rocblas_dgemm_bufferload_limit.yaml]: update

File: Tensile/Tests/extended/bufferload_offset/rocblas_sgemm_bufferload_limit.yaml (hunk 5 above)
Comment (TonyYHsieh) [reply Tensile/Tests/extended/bufferload_offset/rocblas_sgemm_bufferload_limit.yaml]: update

**Comment by TonyYHsieh** [comment]
use another PR for whole feature

**Pull request closed without merging**